### ISSUE TITLE
Add Continuous Integration

### ISSUE THREAD

**Issue by veripoolbot** [verilator/verilator#1468]
---

Author Name: **Todd Strader** (@toddstrader)
Original Redmine Issue: 1468 from https://www.veripool.org

Original Assignee: Todd Strader (@toddstrader)

---

This builds Verilator and runs the entire test suite within Travis-CI:
https://github.com/toddstrader/verilator-dev/tree/travis

I've just been working with the free-for-open-source service via GitHub.  Here's the build that corresponds to HEAD of this branch:
https://travis-ci.com/toddstrader/verilator-dev/builds/115871166

I surveyed the free CI options available on GitHub.  Travis seems widely used, it is easy to set up and as far as I can tell there are no build/CPU-hours/etc. limits for open source projects.  It's pretty simple to get going for your personal fork on GitHub.  Based on experimentation, you seem to be able to launch three workers in parallel for this tier of service.  I'm not currently doing anything with that (see below).  The workers seem to have two CPUs, which I do leverage in the .yml.

The .yml is a bit hairy with the make version upgrade and cpan installation.  I think it might be better to add a ci/ directory to hold helper scripts, but I don't think that's necessary at this point.  I'm open to suggestions on this.

I'm also open to explore (a limited number of) other services if people have suggestions.  But this one seems to work fine.

Future work would include:
* Building for multiple OSes (use parallel workers here)
* Getting the skipped tests to work
* Probably other stuff

I can open issues for these items if we proceed.

Also (and I don't mean to steal any thunder here), I noticed this today:
https://github.com/verilator/verilator

Does this mean that Verilator is going to be mirrored or hosted here?  I was planning to actively sync from veripool.org to my repo to get CI going.  But if this is going to be constantly up-to-date one way or the other, we can easily set it up there as well.  The biggest advantage with that is that incoming PRs will kick off a CI build.  So whenever people get around to looking at PRs, hopefully the tests are already done. 

On a separate note, I got to thinking about this because of the recent Ariane breakage.  I think it would be helpful to build well maintained public Verilog projects head-to-head with Verilator.  That's probably outside the scope of Verilator itself, but would give us more thorough coverage if it can be set up.  I'm not quite sure how that should work, but I'll post something on the forums when I figure something out so people can tune in if they are interested.
